src/System.Management.Automation/help/HelpCommands.cs

```diff
@@ -647,6 +666,48 @@ private void LaunchOnlineHelp(Uri uriToLaunch)
             }
         }
 
+#if !UNIX
+        /// <summary>
+        /// Gets the path to the default browser by querying the Windows registry.
+        /// </summary>
+        /// <returns></returns>
+        private string GetDefaultWebBrowser()
+        {
+            // Check if there is a default browser in the system.
+            const string httpRegkey = @"HKEY_CURRENT_USER\Software\Microsoft\Windows\Shell\Associations\UrlAssociations\http\UserChoice";
+            object progId = Registry.GetValue(httpRegkey, "ProgId", string.Empty);
+            if (progId != null && !string.Empty.Equals(progId))
+            {
+                // Query the registry to find the web browser path.
+                using (RegistryKey browserRegKey = Registry.ClassesRoot.OpenSubKey(progId + "\\shell\\open\\command", false))
+                {
+                    if (browserRegKey != null)
+                    {
+                        const string exeExtension = ".exe";
+                        string browserPath = browserRegKey.GetValue(null).ToString().Replace(/* remove the quotes */ "\"", "");
+                        if (browserPath != null)
+                        {
+                            if (!browserPath.EndsWith(exeExtension, StringComparison.OrdinalIgnoreCase))
+                            {
+                                // Remove any extra chars in the path after ".exe".
+                                browserPath = browserPath.Substring(0, browserPath.LastIndexOf(exeExtension) + exeExtension.Length);
```

Comment:
I know it would be super rare, but what if `LastIndexOf(exeExtension)` returns -1? Maybe change it to something like:
```
int extIndex = browserPath.LastIndexOf(exeExtension);
browserPath = extIndex > 0 ? browserPath.Substring(0, extIndex + exeExtension.Length) : string.Emtpy;
```